dev-packages/electron/native/src/electron-ffmpeg-codecs.c

```diff
@@ -0,0 +1,68 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+// This program must be linked against the ffmpeg library.
+
+#include <stdio.h>
+
+// ffmpeg data
+enum AVMediaType
```

Comment:
Is this original code or copied from somewhere ?

Reply:
It is part of the ffmpeg library itself. Usually they distribute header files that do this kind of declarations (a bit like `.d.ts` files for TypeScript). But here since it is not distributed by Electron, we need to declare it by hand. Although, they originally come from various places (e.g. [`AVCodec`](https://ffmpeg.org/doxygen/trunk/avcodec_8h_source.html#l03476)).

Reply:
Ouch, the linked file is LGPL - I do not think we can copy anything from it. A bit ironic given that we use it to avoid getting into IP trouble :) 

But maybe it's different if it's only interfaces and such, so better ask the Foundation to be sure. Can you remind me on Tuesday, when we're back from long weekend?

Reply:
Well, we use Electron, which already links against libffmpeg, in this case I am just cutting through and directly invoking the library...

But I can remind you Tuesday :)